src/protocols/radius/encode.c

```diff
@@ -255,40 +262,34 @@ static ssize_t encode_tunnel_password(uint8_t *out, size_t outlen,
 	return len;
 }
 
-static ssize_t encode_tlv_hdr_internal(uint8_t *out, size_t outlen,
+static ssize_t encode_tlv_hdr_internal(fr_dbuff_t *dbuff,
 				       fr_da_stack_t *da_stack, unsigned int depth,
 				       fr_cursor_t *cursor, void *encoder_ctx)
 {
 	ssize_t			slen;
-	uint8_t			*p = out;
+	size_t			init_remaining = fr_dbuff_remaining(dbuff);
```

Comment:
I added `FR_DBUFF_MAX_NO_ADVANCE`.  Maybe it'd be better to use that here to initialise a `fr_dbuff_t` on the stack, then we don't have to initialise a new dbuff for every call for encode_tlv_hdr[_internal].

You can then return at the end with:

`return fr_dbuff_advance(dbuff, fr_dbuff_used(tlv))`

Reply:
Thank you. I'll look for places I can use FR_DBUFF_MAX_NO_ADVANCE to advantage.

Reply:
It's been 4+ weeks.  Can these commits be updated to use the new API?

The idea behind DBUFF was to avoid these manual length calculations.

Reply:
Yes, they can, once FR_DBUFF_MAX_NO_ADVANCE() gives you a dbuff with matching start and p.